subprojects/core-api/src/main/java/org/gradle/api/tasks/ClasspathPropertyNormalizer.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks;
+
+import org.gradle.api.Incubating;
+
+/**
+ * Normalizes an input file property that represents a Java runtime classpath.
+ *
+ * Compared to {@link GenericPropertyNormalizer} this normalizer keeps the order of any root files,
+ * but ignores the order and timestamps of files in directories and ZIP/JAR files.
+ *
+ * @see org.gradle.api.tasks.Classpath
+ *
+ * @since 4.3
+ */
+@Incubating
+public interface ClasspathPropertyNormalizer extends PropertyNormalizer {
```

Comment:
Do we want to call this `RuntimeClasspathPropertyNormalizer`? I already forgot again what we discussed last time...

Reply:
Here's our reasoning from before:

> There are two kinds of classpaths currently, so it seems to make sense to qualify both instead of just one of them. However, we don't expect `@CompileClasspath` to be used much outside of Gradle's own codebase: it's specifically designed to handle classpaths consisting of compiled Java code only. For plugin authors, `@Classpath` is going to be the right choice most of the time for classpath inputs.
> 
> Using `@RuntimeClasspath` would suggest there's a decision to be made, where in most cases there isn't. While this is not such a big concern by itself, we agreed that it's okay to leave `@Classpath` be.